### PR TITLE
Add primary key and index to external login user table

### LINKED ISSUE #1655: Cannot delete user with LDAP auth (500 - Internal Server Error)
- Gitea version (or commit ref):
66c803f
- Git version:
git version 2.7.4
- Operating system:
Ubuntu 16.04.2 server LTS x86_64 (ni LXC)
- Database (use `[x]`):
  - [x] PostgreSQL 9.4
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
PostgreSQL complains:
```
2017-05-01 11:36:05.080 CEST [9585] xxx@gitea STATEMENT:  DELETE FROM "external_login_user" WHERE "user_id"=$1
2017-05-01 11:40:28.643 CEST [9585] xxx@gitea ERROR:  Cannot run UPDATE or DELETE on table external_login_user because it does not have a PRIMARY KEY.
```
And very well there is no index on that table:
```
gitea=# select * from pg_indexes where tablename = 'external_login_user';
 schemaname | tablename | indexname | tablespace | indexdef
------------+-----------+-----------+------------+----------
(0 rows)
```

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #1655

I don't know if that's the best way for migration, it would be better to add primary key as SQL commands not recreating table, let me know what you think.

### CODE REVIEW

**Comment by lafriks** [comment]
I created migration this way as there is no way to add primary key on existing table in sqlite3, at least I did not found it

File: models/external_login_user.go [hunk 1]
```diff
@@ -8,9 +8,9 @@ import "github.com/markbates/goth"
 
 // ExternalLoginUser makes the connecting between some existing user and additional external login sources
 type ExternalLoginUser struct {
-	ExternalID    string `xorm:"NOT NULL"`
-	UserID        int64  `xorm:"NOT NULL"`
-	LoginSourceID int64  `xorm:"NOT NULL"`
+	ExternalID    string `xorm:"pk NOT NULL"`
+	UserID        int64  `xorm:"INDEX NOT NULL"`
+	LoginSourceID int64  `xorm:"pk NOT NULL"`
```
Comment (lunny) [inline models/external_login_user.go:13]: So one external user could be only linked to one internal user?

**Comment by lafriks** [comment]
Yes, I think so. When authorizing with external user how would otherwise we know to what internal user to login otherwise. I think I did see error message about it somewhere when translating

**Comment by lunny** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Pull request merged**